txclib/utils.py

```diff
@@ -453,11 +453,37 @@ def color_text(text, color_name, bold=False):
         return text
 
 
-def files_in_project(curpath):
+def _can_walk(base, subdirectories):
     """
-    Iterate over the files in the project.
-    Return each file under ``curpath`` with its absolute name.
+    Determine if the first element of the 'subdirectories' list can be appended
+    to the 'base' path and still specify a directory.
     """
+    # Always consider the last part of the subdirectories the filename
+    if len(subdirectories) < 2:
+        return False
+    # Allow any directory name that doesn't contain the <lang> placeholder
+    return "<lang>" not in subdirectories[0]
+
+
+def get_project_files(curpath, expression):
```

Comment:
We should have some unit tests specifically for this function that will test different cases for curpath and expression arguments.